src/requests/packages.py

```diff
@@ -23,6 +23,5 @@
 target = chardet.__name__
 for mod in list(sys.modules):
     if mod == target or mod.startswith(f"{target}."):
-        target = target.replace(target, "chardet")
```

Comment:
This is unfortunately necessary. This is necessary to support charset-normalizer. The idea here is to import charset if it's installed instead of chardet from this location for users.

This is why we look at `chardet.__name__` (to determine which is most important: https://github.com/psf/requests/blob/839a8edec37c81a18ac8332cfbd44f44e1ae6206/src/requests/packages.py#L8 )

So what we need to do is determine how we want to handle this appropriately.

The idea is that we have two values:

```py
target = "chardet"
target = "charset_normalizer"
```

In the former case if `mod.startswith("chardet.")` then we will be replacing `chardet` with `chardet`. In the latter case, we'd want to replace `charset_normalizer` with `chardet`.

Reply:
Got it, but still the for loop seems necessary. It'll just be mapping the last package of chardet modules to chardet package of requests. 

If that is the expected behaviour then we can just assign the last package without running the for loop.

Reply:
So I'm talking about the intent behind the line, there's obviously a bug in it. And I was trying to leave that as something for you to see for yourself.

The reality is that what we want is 

```py
sys.modules[f"requests.packages.{mod}"] = sys.modules[mod]
mod = mod.replace(target, "chardet")
sys.modules[f"requests.packages.{mod}"] = sys.modules[mod]
```

Because we may as well add charset normalizer in there but also, we want backwards compatibility with chardet

So it's a tiny bit obvious bug and we'd like the intended behavior fixed

Reply:
`sys.modules[f"requests.packages.{mod}"] = sys.modules[mod]
mod = mod.replace(target, "chardet")
sys.modules[f"requests.packages.{mod}"] = sys.modules[mod]`

Doesn't make sense, since then the model will itself change to chardet and will throw a keyerror. 

Have added the support for both chardet and charset_normalizer in the latest commit, with this change - 

`sys.modules[f"requests.packages.{mod}"] = sys.modules[mod]
        target = target.replace(target, "chardet")
        sys.modules[f"requests.packages.{target}"] = sys.modules[mod]`